wasmtime/_table.py

```diff
@@ -1,44 +1,35 @@
 from . import _ffi as ffi
 from ctypes import *
-from wasmtime import TableType, Store, Func, WasmtimeError
+from wasmtime import TableType, Store, WasmtimeError, IntoVal, Val, ValType
 from typing import Optional, Any
 
 
-def get_func_ptr(init: Optional[Func]) -> Optional["pointer[ffi.wasm_func_t]"]:
-    if init is None:
-        return None
-    elif isinstance(init, Func):
-        return init._ptr
-    else:
-        raise TypeError("expected a `Func` or `None`")
-
-
 class Table:
     _ptr: "pointer[ffi.wasm_table_t]"
 
-    def __init__(self, store: Store, ty: TableType, init: Optional[Func]):
+    def __init__(self, store: Store, ty: TableType, init: IntoVal):
         """
         Creates a new table within `store` with the specified `ty`.
-
-        Note that for now only funcref tables are supported and `init` must
-        either be `None` or a `Func`.
         """
 
         if not isinstance(store, Store):
             raise TypeError("expected a `Store`")
         if not isinstance(ty, TableType):
             raise TypeError("expected a `TableType`")
 
-        init_ptr = get_func_ptr(init)
-        ptr = POINTER(ffi.wasm_table_t)()
-        error = ffi.wasmtime_funcref_table_new(store._ptr, ty._ptr, init_ptr, byref(ptr))
-        if error:
-            raise WasmtimeError.__from_ptr__(error)
+        if init is None:
+            init_val = Val.ref_null(ty.element)
+        else:
+            init_val = Val._convert(ty.element, init)
```

Comment:
Technically this branch isn't necessary, right? This is already handled by `_convert`?